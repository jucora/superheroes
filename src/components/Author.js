import React from "react";

const hideInput = () => {
  const input = document.querySelector(".searchHeroInput");
  if (input) {
    input.style.display = "none";
  }
};

const Author = () => {
  hideInput();
  return (
    <div className="author">
      <h1>JULIAN BELMONTE</h1>
      <img src={require("../img/julian.jpg")} alt="Author"></img>
      <div className={"portfolioLink"}>
        <a
          href="https://jucora.github.io/"
          target="_blank"
          rel="noopener noreferrer"
        >
          <h2>Please visit my portfolio</h2>
        </a>
      </div>
    </div>
  );
};

export default Author;
