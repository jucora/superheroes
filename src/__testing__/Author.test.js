import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Author from '../components/Author';

afterEach(cleanup);

configure({ adapter: new Adapter() });
it('should take a snapshot', () => {
  const { asFragment } = render(<Author />);
  expect(asFragment(<Author />)).toMatchSnapshot();
});

it('should contain a div with className author', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('div.author').length).toBe(1);
});

it('should contain a h1', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('h1').length).toBe(1);
});

it('should contain the text: JULIAN BELMONTE inside the h1 tag', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('h1').text()).toBe('JULIAN BELMONTE');
});

it('should contain an img tag', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('img').length).toBe(1);
});

test('the img tag should contain an alt attribute with the value: Author', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('img').prop('alt')).toEqual('Author');
});

test('the img tag should contain a src attribute', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('img').prop('src')).toBeTruthy();
});

it('should contain a div with a className portfolioLink', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('div.portfolioLink')).toBeTruthy();
});

it('should contain an <a> tag to link to Author portfolio page', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('a').length).toBe(1);
});
test('The <a> tag should contain a href', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('a').prop('href')).toBe('https://jucora.github.io/');
});
test('The <a> tag should contain a target _blank', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('a').prop('target')).toBe('_blank');
});

test('The <a> tag should contain a rel attribute with value: noopener noreferrer', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('a').prop('rel')).toBe('noopener noreferrer');
});

it('should contain a h2', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('h2').length).toBe(1);
});

test('The test inside h2 tag should be: Please visit my portfolio', () => {
  const wrapper = shallow(<Author />);
  expect(wrapper.find('h2').text()).toBe('Please visit my portfolio');
});
