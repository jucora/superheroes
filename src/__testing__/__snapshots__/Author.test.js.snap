// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="author"
  >
    <h1>
      JULIAN BELMONTE
    </h1>
    <img
      alt="Author"
      src="julian.jpg"
    />
    <div
      class="portfolioLink"
    >
      <a
        href="https://jucora.github.io/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2>
          Please visit my portfolio
        </h2>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
